ipmikvm

```diff
@@ -303,6 +312,58 @@ install_ikvm_application() {
 }
 # SYNOPSIS: install_ikvm_application JNLP_DATA DESTDIR
 
+find_available_port()
+{
+    local START="$1"
+    local END="$2"
+    local HOSTNAME="${3:-"localhost"}"
+    for P in $(seq $START $END)
+    do
+        if ! nc -z "$HOSTNAME" "$P"
+        then
+            printf "%s" "$P"
+            return
+        fi
+    done
+    echo "Cannot find available local port beetween $START and $END"
+    exit 1
+}
+
+cleanup_ssh()
+{
+    echo "I: Cleaning ssh tunnel"
+    [ "$SSH_PID" ] || return 0
+    kill -9 $SSH_PID
+}
+
+if [ -n "$PROXY" ]; then
+    HTTPS_PORT="$(find_available_port 10443 20443)"
+    HTTP_PORT="$(find_available_port 30080 40080)"
+    VNC_PORT="$(find_available_port 5900 6000)"
+    IPMI_PORT="$(find_available_port 6230 7000)"
+
+    REMOTE_IP=$IP
+    IP=127.0.0.1
+
+    echo "I: Lauching ssh tunnel (-L$IPMI_PORT:$REMOTE_IP:623 -L$VNC_PORT:$REMOTE_IP:5900 -L$HTTPS_PORT:$REMOTE_IP:443 -L$HTTP_PORT:$REMOTE_IP:80 $PROXY)"
+    trap "cleanup_ssh" EXIT TERM INT
+    ssh -N -L$IPMI_PORT:$REMOTE_IP:623 -L$VNC_PORT:$REMOTE_IP:5900 -L$HTTPS_PORT:$REMOTE_IP:443 -L$HTTP_PORT:$REMOTE_IP:80 $PROXY &
```

Comment:
This ssh with the find_available_ports is rather a kludge.

I can see why you added it this way, but it is fragile and takes up quite a bit of code in this script. I wonder if this could be delegated to a separate script instead.